nodes/logic/evolver.py

```diff
@@ -0,0 +1,452 @@
+# ##### BEGIN GPL LICENSE BLOCK #####
+#
+#  This program is free software; you can redistribute it and/or
+#  modify it under the terms of the GNU General Public License
+#  as published by the Free Software Foundation; either version 2
+#  of the License, or (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, write to the Free Software Foundation,
+#  Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+#
+# ##### END GPL LICENSE BLOCK #####
+
+import random
+import time
+from collections import namedtuple
+
+import numpy as np
+import bpy
+from bpy.props import (
+    BoolProperty, StringProperty, EnumProperty, IntProperty, FloatProperty
+    )
+
+from mathutils.noise import seed_set, random
+from sverchok.node_tree import SverchCustomTreeNode
+from sverchok.data_structure import updateNode
+from sverchok.core.update_system import make_tree_from_nodes, do_update
+
+Gene = namedtuple('Gene', 'name g_type, min_n, max_n, range, init_val')
+
+evolver_mem = {}
+
+def is_valid_node(node, genotype_frame):
+    if genotype_frame == 'All' and node.bl_idname == "SvNumberNode":
+        return True
+    if node.parent and node.parent.name == genotype_frame and node.bl_idname == "SvNumberNode":
+        return True
+    return False
+
+def get_genes(target_tree, genotype_frame):
+    genes = []
+    for node in target_tree.nodes:
+        if is_valid_node(node, genotype_frame):
+            num_type = node.selected_mode
+            if node.selected_mode == "float":
+                min_n = node.float_min
+                max_n = node.float_max
+                initial_value = node.float_
+            else:
+                min_n = node.int_min
+                max_n = node.int_max
+                initial_value = node.int_
+            gene = Gene(name=node.name, g_type=num_type, min_n=min_n, max_n=max_n, range=max_n-min_n, init_val=initial_value)
+            genes.append(gene)
+    return genes
+
+class DNA:
+    genes_def = []
+    genes = []
+    fitness = 0
+    def __init__(self, genes_def, random_val=True, empty=False):
+        self.genes_def = genes_def
+        self.genes = []
+        self.fitness = 0
+        if empty:
+            return
+        if random_val:
+            for gene in genes_def:
+                agent_gene = gene.min_n + random() * gene.range
+                if gene.g_type == 'int':
+                    agent_gene = int(agent_gene)
+                self.genes.append(agent_gene)
+        else:
+            for gene in genes_def:
+                agent_gene = gene.init_val
+                self.genes.append(agent_gene)
```

Comment:
It is not very good to do some complex stuff in init method. I would suggest to move this logic in separate method. The name of the method would say what are you trying to do here by itself.